src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -306,15 +307,18 @@ public static int PopCount(uint value)
 
             return SoftwareFallback(value);
 
-            int SoftwareFallback(uint val)
+            int SoftwareFallback(uint v)
             {
-                val = val - ((val >> 1) & 0x_55555555u);
-                val = (val & 0x_33333333u) + ((val >> 2) & 0x_33333333u);
-                val = (val + (val >> 4)) & 0x_0F0F0F0Fu;
-                val = val * 0x_01010101u;
-                val = val >> 24;
+                const uint c1 = 0x_55555555u;
+                const uint c2 = 0x_33333333u;
+                const uint c3 = 0x_0F0F0F0Fu;
+                const uint c4 = 0x_01010101u;
 
-                return (int)val;
+                v = v - ((v >> 1) & c1);
+                v = (v & c2) + ((v >> 2) & c2);
+                v = (((v + (v >> 4)) & c3) * c4) >> 24;
```

Comment:
Is this method that uses multiplication actually fastest on the current hardware?

Reply:
I am not familiar with any other method - what is the alternative?

Reply:
Oh you mean a lookup table?

Reply:
http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel

Reply:
It looks like the `parallel` version is **1.24x** slower than `multiply`
```
 Method |     Mean |     Error |    StdDev | Ratio | RatioSD |
------- |---------:|----------:|----------:|------:|--------:|
    Mul | 3.608 ns | 0.0558 ns | 0.0522 ns |  1.00 |    0.00 |
    Par | 4.474 ns | 0.0385 ns | 0.0360 ns |  1.24 |    0.02 |
```
Here's the implementation I used:
```csharp
int SoftwareFallback(uint v)
{
    v = v - ((v >> 1) & 0x55555555u);
    v = ((v >> 2) & 0x33333333u) + (v & 0x33333333u);
    v = ((v >> 4) + v) & 0x0F0F0F0Fu;
    v = ((v >> 8) + v) & 0x00FF00FFu;
    v = ((v >> 16) + v) & 0x0000FFFFu;

    return (int)v;
}
```

Reply:
ok, thanks for checking

Reply:
It's worth noting that [Counting bits set, in parallel](http://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetParallel) actually gives both `Mul` and `Par` as implementations.

It first lists `Par` as the "general-purpose" algorithm and then lists `Mul` as the "best method for counting bits in a 32-bit integer" and that it is a hybridized approach between `Par` and the more complicated algorithm that uses 64-bit multiplication.